daniel-planets.rb

```diff
@@ -0,0 +1,167 @@
+planets = [
+  {name: "Mercury",index_from_sun: 1,distance_from_sun: 0.4,
+   mass: 0.06,date_discovered: "2nd millenium B.C.E.",
+   number_of_moons: 0,days_in_a_year: 87.97},
+
+  {name: "Venus",index_from_sun: 2,distance_from_sun: 0.7,
+   mass: 0.82,date_discovered: "2nd millenium B.C.E.",
+   number_of_moons: 0,days_in_a_year: 224.7},
+
+  {name: "Earth",index_from_sun: 3,distance_from_sun: 1,
+   mass: 1,date_discovered: "......",
+   number_of_moons: 1,days_in_a_year: 365.26},
+
+  {name: "Mars",index_from_sun: 4,distance_from_sun: 1.5,
+   mass: 0.11,date_discovered: "2nd millenium B.C.E.",
+   number_of_moons: 2,days_in_a_year: 686.98},
+
+  {name: "Jupiter",index_from_sun: 5,distance_from_sun: 5.2,
+   mass: 317.8,date_discovered: "2nd millenium B.C.E.",
+   number_of_moons: 67,days_in_a_year: 4332.82},
+
+  {name: "Saturn",index_from_sun: 6,distance_from_sun: 9.5,
+   mass: 95.2,date_discovered: "2nd millenium B.C.E.",
+   number_of_moons: 62,days_in_a_year: 10755.7},
+
+  {name: "Uranus",index_from_sun: 7,distance_from_sun: 19.2,
+   mass: 14.6,date_discovered: "April 26, 1781",
+   number_of_moons: 27,days_in_a_year: 30687.15},
+
+  {name: "Neptune",index_from_sun: 8,distance_from_sun: 30.1,
+   mass: 17.2,date_discovered: "January 24, 1801",
+   number_of_moons: 14,days_in_a_year: 60190.03}
+]
+
+class Planet #now attributes can come from constructor
+  attr_accessor :name,
+                :index_from_sun,
+                :distance_from_sun, #in AU ('Astronomical Units')
+                :mass, #relative to mass of Earth
+                :date_discovered,
+                :number_of_moons,
+                :days_in_a_year #Earth days
+
+  def initialize(p)
+    @name = p[:name]
+    @index_from_sun = p[:index_from_sun]
+    @distance_from_sun = p[:distance_from_sun]
+    @mass = p[:mass]
+    @date_discovered = p[:date_discovered]
+    @number_of_moons = p[:number_of_moons]
+    @days_in_a_year = p[:days_in_a_year]
+  end
+end
+
+all_planets = planets.map {|planet_hash| Planet.new planet_hash}
+#what happens if we instantiate but don't assign to a variable?
```

Comment:
If you instantiate but don't assign to a variable (or don't add it to a collection) is that it's immediately lost, it still creates it but it will immediately thrown away. If you do assign it to a collection like you are here, then it doesn't matter that you didn't assign it to a variable. Nothing is different.